classes/schema/ActionScheduler_StoreSchema.php

```diff
@@ -55,7 +55,8 @@ protected function get_table_definition( $table ) {
 				        KEY args (args($max_index_length)),
 				        KEY group_id (group_id),
 				        KEY last_attempt_gmt (last_attempt_gmt),
-				        KEY claim_id (claim_id)
+				        KEY claim_id (claim_id),
+						KEY `claim_id, status, scheduled_date_gmt` (`claim_id`, `status`, `scheduled_date_gmt`)
```

Comment:
For some strange reason, i'm unable to edit this PR. Possibly because I've deleted the branch? I see no reason not too change it to what you suggested, although I've not seen any similar key names.

```KEY `claim_id-status-scheduled_date_gmt` (`claim_id`, `status`, `scheduled_date_gmt`)```

Regarding your findings, it's strange how you can;'t reproduce this. I assume your tables are in innodb? if so, it could be something in the server settings. In any case, we've had this index for nearly a year an never had deadlocks

Reply:
Thanks, I guess I can update the index name during the merge.